### PR TITLE
fix(vector): Toggle both service and headless service with service.enabled

### LINKED ISSUE #100: vector chart: vector-headless service is still created when service.enable=false
headless service is still created when service is disabled in values. 

service-headless' template condition is [set](https://github.com/vectordotdev/helm-charts/blob/6fcb9de045f2c0b0b01bf99082413b7574114ed1/charts/vector/templates/service-headless.yaml#L1) to `{{- if or (eq .Values.role "Aggregator") (eq .Values.role "Stateless-Aggregator") -}}` 

while service's template [has](https://github.com/vectordotdev/helm-charts/blob/6fcb9de045f2c0b0b01bf99082413b7574114ed1/charts/vector/templates/service.yaml#L1) `{{- if and .Values.service.enabled }}`

Possible solution:
```yaml
{{- if and .Values.service.enabled (or (eq .Values.role "Aggregator") (eq .Values.role "Stateless-Aggregator")) -}}
```

Use case for having service completely disabled: 

using Vector for getting AWS ALB logs from S3 bucket to Loki or any other pull configuration.

### PR DESCRIPTION
**Pull request by spencergilbert**
Signed-off-by: Spencer Gilbert <spencer.gilbert@datadoghq.com>

Closes #100

This mirrors the condition used for both the service and service-headless.
Headless services are generally more useful for load balancing or being able
to select specific instances from a StatefulSet, but rather than introducing
more configuration this seems simpler unless a user requests more specific options.

### CODE REVIEW

**Review by jszwedko — COMMENTED** [review]
What's the difference between the headless and head(ed?) service?

**Comment by spencergilbert** [comment]
> What's the difference between the headless and head(ed?) service?

https://kubernetes.io/docs/concepts/services-networking/service/#headless-services

tl;dr: `For headless Services, a cluster IP is not allocated, kube-proxy does not handle these Services, and there is no load balancing or proxying done by the platform for them.`

For a concrete example, HAProxy in our chart uses the headless service to do DNS resolutions for each vector instance so it can create it's backends.

File: charts/vector/templates/service-headless.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if or (eq .Values.role "Aggregator") (eq .Values.role "Stateless-Aggregator") -}}
+{{- if and .Values.service.enabled }}
```
Comment (jszwedko) [inline charts/vector/templates/service-headless.yaml:1]: It seems like the `and` here is superfluous?

File: charts/vector/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Vector
 
-![Version: 0.1.0-beta.2](https://img.shields.io/badge/Version-0.1.0--beta.2-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 0.17.3-distroless-libc](https://img.shields.io/badge/AppVersion-0.17.3--distroless--libc-informational?style=flat-square)
+![Version: 0.1.0-beta.3](https://img.shields.io/badge/Version-0.1.0--beta.3-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 0.17.3-distroless-libc](https://img.shields.io/badge/AppVersion-0.17.3--distroless--libc-informational?style=flat-square)
```
Comment (jszwedko) [inline charts/vector/README.md:3]: Still wondering if this'll conflict with https://github.com/vectordotdev/helm-charts/pull/104

**Review by jszwedko — APPROVED** [review]
Looks good! Thanks again for explaining offline (but online).

File: charts/vector/templates/service-headless.yaml (hunk 1 above)
Comment (spencergilbert) [reply charts/vector/templates/service-headless.yaml]: Huh - I put the `and` in the other template too 🤔
